[CES-985]: Executive Spell | Aug 24th 2022

hbarcelos opened the pull request:
- chore: cleanup before 2022-08-24 spell
- feat: add RWA009_A_URN draw-down for this week

# Description

# Contribution Checklist

- [ ] PR title starts with `(PE-<TICKET_NUMBER>)`
- [ ] Code approved
- [ ] Tests approved
- [ ] CI Tests pass

# Checklist

- [ ] Every contract variable/method declared as public/external private/internal
- [ ] Consider if this PR needs the `officeHours` modifier override
- [ ] Verify expiration (`30 days` unless otherwise specified)
- [ ] Verify hash in the description matches [here](https://emn178.github.io/online-tools/keccak_256.html)
- [ ] Validate all addresses used are in changelog or known
- [ ] Notify any external teams affected by the spell so they have the opportunity to review
- [ ] Deploy spell `ETH_GAS="XXX" ETH_GAS_PRICE="YYY" make deploy`
- [ ] Verify `mainnet` contract on etherscan
- [ ] Change test to use mainnet spell address and deploy timestamp
- [ ] Run `make archive-spell` or `make date="YYYY-MM-DD" archive-spell` to make an archive directory and copy `DssSpell.sol`, `DssSpell.t.sol`, `DssSpell.t.base.sol`, and `DssSpellCollateralOnboarding.sol`
- [ ] `squash and merge` this PR

iamchrissmith commented on src/DssSpell.sol:
```diff
@@ -27,36 +27,18 @@ interface RwaUrnLike {
-    uint256 public constant MILLION            = 10 **  6;
-    uint256 public constant BILLION            = 10 **  9;
-    uint256 public constant WAD                = 10 ** 18;
+    uint256 public constant MILLION = 10 **  6;
+    uint256 public constant BILLION = 10 **  9;
```
```suggestion
```

not used

iamchrissmith commented on src/DssSpell.sol:
```diff
@@ -68,101 +50,20 @@ contract DssSpellAction is DssAction {
+        // https://vote.makerdao.com/polling/QmQMDasC#poll-detail
+        // Weekly Draw for HVB
+        address RWA009_A_URN = DssExecLib.getChangelogAddress("RWA009_A_URN");
+        // Draw again for Aug 24th Exec Draw
```
```suggestion
```

Probably don't need this one

iamchrissmith commented on src/DssSpell.t.sol:
```diff
@@ -541,17 +541,26 @@ contract DssSpellTest is DssSpellTestBase {
-        assertEq(rwaconduitout_008.can(RWA008_A_OPERATOR), 0, "RWA008: [pre-spell] bad outputConduit.can(operator)");
+    function testRWA009_SPELL_DRAW() public {
+        address rwaUrn009       = addr.addr("RWA009_A_URN");
+        address rwaUrn009Output = addr.addr("RWA009_A_OUTPUT_CONDUIT"); // for goerli, we use the pause proxy
```
```suggestion
        address rwaUrn009Output = addr.addr("RWA009_A_OUTPUT_CONDUIT"); 
```

incorrect comment for mainnet

iamchrissmith commented on src/DssSpell.t.sol:
```diff
@@ -541,17 +541,26 @@ contract DssSpellTest is DssSpellTestBase {
 
-        assertEq(rwaconduitout_008.can(RWA008_A_OPERATOR), 1, "RWA008: [post-spell] bad outputConduit.can(operator)");
-    }
+        // Check if spell draw 25mm DAI to Output Conduit (Pause Proxy)
```
```suggestion
        // Check if spell draw 25mm DAI to Output Conduit
```

copy/paste from goerli

iamchrissmith commented on src/DssSpell.t.sol:
```diff
@@ -541,17 +541,26 @@ contract DssSpellTest is DssSpellTestBase {
-        assertEq(rwaconduitout_008.can(RWA008_A_OPERATOR), 1, "RWA008: [post-spell] bad outputConduit.can(operator)");
-    }
+        // Check if spell draw 25mm DAI to Output Conduit (Pause Proxy)
+        assertEq(dai.balanceOf(address(rwaUrn009Output)), prevBalance + drawAmount, "RWA009/dai-drawn-was-not-send-to-the-recipient");
```
```suggestion
        assertEq(dai.balanceOf(address(rwaUrn009Output)), prevBalance + drawAmount, "RWA009/dai-drawn-was-not-sent-to-the-recipient");
```

iamchrissmith reviewed (changes requested):
some initial changes/comments before we get the hash

iamchrissmith commented on src/DssSpell.sol:
```diff
@@ -27,36 +27,16 @@ interface RwaUrnLike {
-    uint256 public constant MILLION            = 10 **  6;
-    uint256 public constant BILLION            = 10 **  9;
-    uint256 public constant WAD                = 10 ** 18;
+    uint256 public constant WAD     = 10 ** 18;
```
Should probably remove the line after this and then fix the spacing so the `=` line up (or just remove the extra spacing between `WAD` and `=` here

hbarcelos replied:
Done

iamchrissmith commented:
minor change, code looks good, tests pass:

```rust
Running 1 test for src/test/starknet.t.sol:StarknetTests
[PASS] testStarknet() (gas: 737401)
Test result: ok. 1 passed; 0 failed; finished in 17.81s

Running 15 tests for src/DssSpell.t.sol:DssSpellTest
[PASS] testCastCost() (gas: 545320)
[PASS] testFailTooEarly() (gas: 401885)
[PASS] testFailTooLate() (gas: 401873)
[PASS] testFailWrongDay() (gas: 401803)
[PASS] testFail_notScheduled() (gas: 14147)
[PASS] testOnTime() (gas: 541709)
[PASS] testRWA009_SPELL_DRAW() (gas: 562941)
[PASS] testSpellIsCast_GENERAL() (gas: 22568954)
[PASS] test_auth() (gas: 9223336852497263829)
[PASS] test_auth_in_sources() (gas: 9223336852484867045)
[PASS] test_bytecode_matches() (gas: 1113115)
[PASS] test_chainlog_values() (gas: 5831502)
[PASS] test_chainlog_version_bump() (gas: 3078589)
[PASS] test_nextCastTime() (gas: 404902)
[PASS] test_use_eta() (gas: 337923)
```

iamchrissmith reviewed (approved):
last change lgtm

tests still passing:

```rust
Running 1 test for src/test/starknet.t.sol:StarknetTests
[PASS] testStarknet() (gas: 737056)
Test result: ok. 1 passed; 0 failed; finished in 17.36s

Running 15 tests for src/DssSpell.t.sol:DssSpellTest
[PASS] testCastCost() (gas: 544975)
[PASS] testFailTooEarly() (gas: 401885)
[PASS] testFailTooLate() (gas: 401873)
[PASS] testFailWrongDay() (gas: 401803)
[PASS] testFail_notScheduled() (gas: 14147)
[PASS] testOnTime() (gas: 541364)
[PASS] testRWA009_SPELL_DRAW() (gas: 562596)
[PASS] testSpellIsCast_GENERAL() (gas: 22568609)
[PASS] test_auth() (gas: 9223336852497263829)
[PASS] test_auth_in_sources() (gas: 9223336852484867045)
[PASS] test_bytecode_matches() (gas: 1113115)
[PASS] test_chainlog_values() (gas: 5831157)
[PASS] test_chainlog_version_bump() (gas: 3078244)
[PASS] test_nextCastTime() (gas: 404902)
[PASS] test_use_eta() (gas: 337923)

```

brianmcmichael reviewed (approved):
Code looks good. Tests pass.

```haskell
dapp-test: rpc block: latest
Running 15 tests for src/DssSpell.t.sol:DssSpellTest
[PASS] testFail_notScheduled() (gas: 7147)
[PASS] test_auth_in_sources() (gas: 281473905229126)
[PASS] test_use_eta() (gas: 330123)
[PASS] testOnTime() (gas: 554262)
[PASS] test_nextCastTime() (gas: 393502)
[PASS] test_chainlog_values() (gas: 5181555)
[PASS] testRWA009_SPELL_DRAW() (gas: 562994)
[PASS] testFailTooEarly() (gas: 370185)
[PASS] test_bytecode_matches() (gas: 1101615)
[PASS] test_chainlog_version_bump() (gas: 3087142)
[PASS] test_auth() (gas: 281473917626417)
[PASS] testSpellIsCast_GENERAL() (gas: 20443007)
[PASS] testFailWrongDay() (gas: 370103)
[PASS] testCastCost() (gas: 555873)
[PASS] testFailTooLate() (gas: 370145)

Running 1 tests for src/test/starknet.t.sol:StarknetTests
[PASS] testStarknet() (gas: 727454)
```

brianmcmichael commented on src/test/config.sol:
```diff
@@ -98,10 +98,10 @@ contract Config {
-            deployed_spell_created:         1660252416,                 // use get-created-timestamp.sh if deployed
-            previous_spell:                 address(0x67A45c2163798d47c32Fc2bcD5Dc0ABc6dCDFe78),        // supply if there is a need to test prior to its cast() function being called on-chain.
-            office_hours_enabled:           false,              // true if officehours is expected to be enabled in the spell
+            deployed_spell:                 address(0),        // populate with deployed spell if deployed
```
```suggestion
            deployed_spell:                 address(0xb18a078aafd872fe06a2ba8c5a7cc068d6bd79a9),        // populate with deployed spell if deployed
```

iamchrissmith replied:
checksum address: `0xB18a078AAFD872fe06A2BA8c5A7CC068D6bD79a9`

brianmcmichael commented on src/test/config.sol:
```diff
@@ -98,10 +98,10 @@ contract Config {
-            previous_spell:                 address(0x67A45c2163798d47c32Fc2bcD5Dc0ABc6dCDFe78),        // supply if there is a need to test prior to its cast() function being called on-chain.
-            office_hours_enabled:           false,              // true if officehours is expected to be enabled in the spell
+            deployed_spell:                 address(0),        // populate with deployed spell if deployed
+            deployed_spell_created:         0,                 // use get-created-timestamp.sh if deployed
```
```suggestion
            deployed_spell_created:         1661290827,                 // use get-created-timestamp.sh if deployed
```

iamchrissmith commented:
added corrected address to avoid the checksum error. but tests pass with address/timestamp added to config.  Should be good to archive and get final approval before publishing to govalpha

hbarcelos commented:
- Updated `config.sol` with the deployed spell and timestamp.
- Added the current spell to the archive.

Tests are passing:

```haskell
Running 15 tests for src/DssSpell.t.sol:DssSpellTest
[PASS] testFail_notScheduled() (gas: 12147)
[PASS] test_auth_in_sources() (gas: 281473905229157)
[PASS] test_use_eta() (gas: 337123)
[PASS] testOnTime() (gas: 563919)
[PASS] test_nextCastTime() (gas: 400502)
[PASS] test_chainlog_values() (gas: 5193212)
[PASS] testRWA009_SPELL_DRAW() (gas: 572651)
[PASS] testFailTooEarly() (gas: 377078)
[PASS] test_bytecode_matches() (gas: 1111115)
[PASS] test_chainlog_version_bump() (gas: 3098799)

[PASS] test_auth() (gas: 281473917626448)
[PASS] testSpellIsCast_GENERAL() (gas: 21552394)
[PASS] testFailWrongDay() (gas: 377103)
[PASS] testCastCost() (gas: 565530)
[PASS] testFailTooLate() (gas: 377173)

Running 1 tests for src/test/starknet.t.sol:StarknetTests
[PASS] testStarknet() (gas: 737111)
```

brianmcmichael reviewed (approved):
Archive and tests with spell work out.

```haskell
dapp-test: rpc block: latest
Running 15 tests for src/DssSpell.t.sol:DssSpellTest
[PASS] testFail_notScheduled() (gas: 12147)
[PASS] test_auth_in_sources() (gas: 281473905229157)
[PASS] test_use_eta() (gas: 337123)
[PASS] testOnTime() (gas: 563459)
[PASS] test_nextCastTime() (gas: 400502)
[PASS] test_chainlog_values() (gas: 5192752)
[PASS] testRWA009_SPELL_DRAW() (gas: 572191)
[PASS] testFailTooEarly() (gas: 377078)
[PASS] test_bytecode_matches() (gas: 1111115)
[PASS] test_chainlog_version_bump() (gas: 3098339)
[PASS] test_auth() (gas: 281473917626448)
[PASS] testSpellIsCast_GENERAL() (gas: 21551934)
[PASS] testFailWrongDay() (gas: 377103)
[PASS] testCastCost() (gas: 565070)
[PASS] testFailTooLate() (gas: 377173)

Running 1 tests for src/test/starknet.t.sol:StarknetTests
[PASS] testStarknet() (gas: 736651)
```

iamchrissmith reviewed (changes requested):
can you change the directory name for the archive to be just `2022-08-24-DssSpell` without the duplicate `-DssSpell` (note the `make archive-spell` command takes just the date not the name of the directory)

iamchrissmith commented:
Tests pass, diff check works out on verified spell and archive contents are good for me.

```rust
Running 1 test for src/test/starknet.t.sol:StarknetTests
[PASS] testStarknet() (gas: 737411)
Test result: ok. 1 passed; 0 failed; finished in 18.20s

Running 15 tests for src/DssSpell.t.sol:DssSpellTest
[PASS] testCastCost() (gas: 545330)
[PASS] testFailTooEarly() (gas: 401778)
[PASS] testFailTooLate() (gas: 401873)
[PASS] testFailWrongDay() (gas: 401803)
[PASS] testFail_notScheduled() (gas: 14147)
[PASS] testOnTime() (gas: 541719)
[PASS] testRWA009_SPELL_DRAW() (gas: 562951)
[PASS] testSpellIsCast_GENERAL() (gas: 23668694)
[PASS] test_auth() (gas: 9223336852497263829)
[PASS] test_auth_in_sources() (gas: 9223336852484867045)
[PASS] test_bytecode_matches() (gas: 1113115)
[PASS] test_chainlog_values() (gas: 5831512)
[PASS] test_chainlog_version_bump() (gas: 3078599)
[PASS] test_nextCastTime() (gas: 404902)
[PASS] test_use_eta() (gas: 337923)
```

We are good to **publish to GovAlpha** but we need to fix the archive directory before we **merge the PR**

iamchrissmith reviewed (approved):
archive directory looks good

brianmcmichael reviewed (approved):
Good to merge.
